src/group/stabchain/builder/ift.rs

```diff
@@ -135,13 +136,18 @@ impl<T: MovedPointSelector> StabchainBuilderIFT<T> {
         // Store the updated record in the chain
         self.chain[self.current_pos] = record;
     }
+}
 
-    pub(super) fn extend(&mut self, p: Permutation) {
+impl<M> super::Builder for StabchainBuilderIFT<M>
+where
+    M: MovedPointSelector,
+{
+    fn extend(&mut self, p: Permutation) {
```

Comment:
My only comment would be that not all algorithms work with an extend method, so this probably need to be more general. For example with the random generation, there are two separate stages for a strong generating set construction, and then a test for the strong generation set.

Reply:
No worries, I'll refactor with a set_generators method that should be generic enough for any of our needs

Reply:
I've pushed the fix, let me know what you think

Reply:
Yeah looks good to me now. I've approved